src/lib/registration/sw_mgmt.rb

```diff
@@ -170,7 +170,7 @@ def self.find_base_product
       # just for debugging:
       return FAKE_BASE_PRODUCT if ENV["FAKE_BASE_PRODUCT"]
 
-      return online_base_product if Y2Packager::MediumType.online?
+      return online_base_product if Stage.initial && Y2Packager::MediumType.online?
```

Comment:
are you sure that here it is proper place? I expect fix to be in MediumType so it does not crash outside insts-sys.

Reply:
Actually the [exception](https://github.com/yast/yast-packager/blob/master/src/lib/y2packager/medium_type.rb#L86) is thrown there deliberately. Sure, the code in `Y2Packager::MediumType.online?` could return `false` (or `nil`) when not running in inst-sys. But that could hide the problem and the code could behave unexpectedly in some cases. It would be also difficult to debug. So I decided to make it fail verbosely.